flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBProperty.java

```diff
@@ -31,55 +31,92 @@
+    CurSizeAllMemTables("cur-size-all-mem-tables", PropertyType.NUMBER),
+    SizeAllMemTables("size-all-mem-tables", PropertyType.NUMBER),
+    NumEntriesActiveMemTable("num-entries-active-mem-table", PropertyType.NUMBER),
+    NumEntriesImmMemTables("num-entries-imm-mem-tables", PropertyType.NUMBER),
+    NumDeletesActiveMemTable("num-deletes-active-mem-table", PropertyType.NUMBER),
+    NumDeletesImmMemTables("num-deletes-imm-mem-tables", PropertyType.NUMBER),
+    EstimateNumKeys("estimate-num-keys", PropertyType.NUMBER),
+    EstimateTableReadersMem("estimate-table-readers-mem", PropertyType.NUMBER),
+    NumSnapshots("num-snapshots", PropertyType.NUMBER),
+    NumLiveVersions("num-live-versions", PropertyType.NUMBER),
+    EstimateLiveDataSize("estimate-live-data-size", PropertyType.NUMBER),
+    TotalSstFilesSize("total-sst-files-size", PropertyType.NUMBER),
+    LiveSstFilesSize("live-sst-files-size", PropertyType.NUMBER),
+    EstimatePendingCompactionBytes("estimate-pending-compaction-bytes", PropertyType.NUMBER),
+    NumRunningCompactions("num-running-compactions", PropertyType.NUMBER),
+    NumRunningFlushes("num-running-flushes", PropertyType.NUMBER),
+    ActualDelayedWriteRate("actual-delayed-write-rate", PropertyType.NUMBER),
+    IsWriteStopped("is-write-stopped", PropertyType.NUMBER),
+    BlockCacheCapacity("block-cache-capacity", PropertyType.NUMBER),
+    BlockCacheUsage("block-cache-usage", PropertyType.NUMBER),
+    BlockCachePinnedUsage("block-cache-pinned-usage", PropertyType.NUMBER),
+    NumFilesAtLevel("num-files-at-level", PropertyType.STRING);
 
     private static final String ROCKS_DB_PROPERTY_FORMAT = "rocksdb.%s";
 
     private static final String CONFIG_KEY_FORMAT = "state.backend.rocksdb.metrics.%s";
 
-    private final String property;
+    private final String propertyName;
 
-    RocksDBProperty(String property) {
-        this.property = property;
+    private final PropertyType type;
+
+    /** Property type. */
+    public enum PropertyType {
+        NUMBER,
+        STRING
     }
 
-    /**
-     * @return property string that can be used to query {@link
-     *     RocksDB#getLongProperty(ColumnFamilyHandle, String)}.
-     */
-    public String getRocksDBProperty() {
-        return String.format(ROCKS_DB_PROPERTY_FORMAT, property);
+    RocksDBProperty(String propertyName, PropertyType type) {
+        this.propertyName = propertyName;
+        this.type = type;
+    }
+
+    public String getPropertyName() {
+        return this.propertyName;
+    }
+
+    public static RocksDBProperty getRocksDBProperty(final String property) {
+        // NumFilesAtLevel controls multiple levels of file count monitoring, each a separate metric
+        if (property.startsWith(NumFilesAtLevel.getPropertyName())) {
+            return NumFilesAtLevel;
+        }
+        for (final RocksDBProperty rocksDBProperty : RocksDBProperty.values()) {
```

Comment:
Is it inefficient? 
Actually I would prefer `NumFilesAtLevel0` to `NumFilesAtLevel6` as separate values of this enum, and use RocksDBProperty instance itself instead of the property name everywhere. WDYT? @masteryhx

Reply:
@Zakelly @masteryhx It sounds reasonable and I have modified it as suggested.